server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -185,6 +190,19 @@ public ObjectMapper build(MapperBuilderContext context) {
                 buildMappers(context.createChildContext(name))
             );
         }
+
+        @Override
+        public final ObjectMapper build(MapperBuilderContext context) {
+            if (objectMapperCache != null) {
+                return objectMapperCache;
+            }
+            objectMapperCache = doBuild(context);
+            return objectMapperCache;
+        }
+
+        protected void clearObjectMapperCache() {
+            objectMapperCache = null;
+        }
```

Comment:
At this point, I am not sure which is better. Doing this, or reverting: https://github.com/elastic/elasticsearch/pull/90674

@javanna ^

Reply:
Thanks for pinging. I think reverting for now is our wisest option. Happy you have come to a similar conclusion, I think. Long term, we should reconsider what we want to do . Perhaps the updates to the benchmarks is something we do want to get in separately?

Reply:
>  Perhaps the updates to the benchmarks is something we do want to get in separately?

Yes, I will open a separate PR. Its a small adjustment, but showed how we still hadn't addressed the performance issue.